app/src/main/java/app/vdh/org/vdhapp/feature/report/data/common/repository/ReportRepositoryImpl.kt

```diff
@@ -0,0 +1,105 @@
+package app.vdh.org.vdhapp.feature.report.data.common.repository
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.map
+import app.vdh.org.vdhapp.core.helpers.CallResult
+import app.vdh.org.vdhapp.feature.report.data.common.local.ReportDao
+import app.vdh.org.vdhapp.feature.report.data.common.local.entity.ReportEntity
+import app.vdh.org.vdhapp.feature.report.data.common.remote.ObservationApiClient
+import app.vdh.org.vdhapp.feature.report.data.common.remote.dto.ObservationDto
+import app.vdh.org.vdhapp.feature.report.data.common.remote.dto.ObservationListDto
+import app.vdh.org.vdhapp.feature.report.data.common.remote.dto.toReportEntities
+import app.vdh.org.vdhapp.core.helpers.safeCall
+import app.vdh.org.vdhapp.feature.report.domain.common.model.ReportModel
+import app.vdh.org.vdhapp.feature.report.domain.common.model.Status
+import app.vdh.org.vdhapp.feature.report.domain.common.repository.ReportRepository
+import java.util.concurrent.TimeUnit
+
+class ReportRepositoryImpl(
+    private val reportDao: ReportDao,
+    private val observationApiClient: ObservationApiClient,
+    private val appContext: Context
+) : ReportRepository {
+
+    override suspend fun getReports(hoursAgo: Int, status: Status?): LiveData<List<ReportModel>> {
+        val from = System.currentTimeMillis() - TimeUnit.HOURS.toMillis(hoursAgo.toLong())
```

Comment:
What about timezones ? :D

Reply:
Time what ? :p ! but you're right opening an issue thx